engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcessTest.scala

```diff
@@ -0,0 +1,159 @@
+package pl.touk.nussknacker.engine.canonicalgraph
+
+import org.scalatest.{FunSuite, Matchers}
+import pl.touk.nussknacker.engine.api.{MetaData, StreamMetaData}
+import pl.touk.nussknacker.engine.canonicalgraph.canonicalnode.{CanonicalNode, Case, FilterNode, FlatNode, SplitNode, Subprocess, SwitchNode}
+import pl.touk.nussknacker.engine.graph.exceptionhandler.ExceptionHandlerRef
+import pl.touk.nussknacker.engine.graph.node.{Filter, Sink, Source, Split, SubprocessInput, Switch}
+import pl.touk.nussknacker.engine.graph.sink.SinkRef
+import pl.touk.nussknacker.engine.graph.source.SourceRef
+import pl.touk.nussknacker.engine.graph.subprocess.SubprocessRef
+
+class CanonicalProcessTest extends FunSuite with Matchers {
+  import pl.touk.nussknacker.engine.spel.Implicits.asSpelExpression
+
+  val source1 = FlatNode(Source("in", SourceRef("sourceType", Nil)))
+
+  val sink1 = FlatNode(Sink("out", SinkRef("sinkType", Nil)))
+
+  val disabledFilter1 = FilterNode(data = Filter("filter1", "''", isDisabled = Some(true)), List(sink1))
+
+  def subprocess(output: List[CanonicalNode], isDisabled: Boolean) =
+    Subprocess(
+      SubprocessInput(
+        "sub1",
+        SubprocessRef("sub1", Nil),
+        isDisabled = Some(isDisabled)
+      ),
+      Map("subOut" -> output)
```

Comment:
Can you provide test for subprocess with many outputs?

Reply:
:+1: